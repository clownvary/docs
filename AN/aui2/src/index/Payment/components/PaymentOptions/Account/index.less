@import '~shared/style/variables/index.less';

@input-width: 112px;

.refund-account {
  &__request {
    margin-left: 30px;
  }

  .input-group {
    width: @input-width;
  }

  .aaui-flexbox {
    justify-content: space-between;
  }

  & &-request {
    align-items: flex-start;
    justify-content: flex-start;
    padding: 21px;
    background-color: @state-info-bg;

    &__dropdown {
      width: 200px;
    }

    &__label {
      padding-right: 20px;
      font-weight: @headings-font-weight;
    }

    &__other {
      width: 420px;
      height: 120px;
      padding: 8px;
      margin-top: 15px;
      border: solid 1px #ccc;
      border-radius: 3px;

      &--error {
        color: @state-danger-text;
        background-color: @state-danger-bg;
        border-color: @state-danger-border;
      }
    }

    &__content {
      display: flex;
      flex-direction: column;
    }
  }
}
