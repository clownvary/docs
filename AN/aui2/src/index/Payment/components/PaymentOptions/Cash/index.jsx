import React from 'react';
import { connect } from 'react-redux';
import classNames from 'classnames';
import InputNumeric from 'react-base-ui/lib/components/InputNumeric';

import UIComponent from 'shared/components/UIComponent';
import Dropdown from 'react-base-ui/lib/components/Dropdown';
import { confirm } from 'react-base-ui/lib/services/dialog';

import calculate from './calculate';
import { changeCashAmount, calculateCashChange } from '../../../actions/paymentOptions/cash';
import { changeRemaining } from '../../../actions';
import formatCurrency from '../../../utils/formatCurrency';
import { getRemaining, getDefaultAmount } from '../utils/payment';
import checkError from '../utils/checkError';
import AMIds from '../../../automationIds';
import { fieldProps } from '../../../consts';

import './index.less';

let confirmCallback = () => { };

/* eslint-disable class-methods-use-this */
export class Cash extends UIComponent {
  constructor(props) {
    super(props);
    this.state = { alertMessage: '' };
  }

  render() {
    const { item, value, index, children, optionLen,
      isRefund, hasManuallySplitOption } = this.props;
    const hasError = checkError(item);
    const roundValue = this.roundAmount(item.amount);

    return (
      <div className="payment-option payment-cash">
        <div
          className={classNames(
            'payment-option-list',
            'aaui-flexbox',
            { 'payment-option-list--error': hasError }
          )}
        >
          <Dropdown
            data={item.list}
            value={value}
            data-qa-id={AMIds.paymentOption.optionList}
            onChange={({ value: val }) => { this.props.onChange(val); }}
            disabled={index !== optionLen - 1 && hasManuallySplitOption}
          />
          <div className="aaui-flexbox">
            <label className="payment-symbol" htmlFor="cashAmount">$</label>
            <InputNumeric
              id="cashAmount"
              value={roundValue}
              ref={(inputNumeric) => { this._amountInput = inputNumeric; }}
              data-qa-id={AMIds.paymentOption.optionAmount}
              min={0}
              onBlur={() => this.onBlurCashInput(hasError)}
            />
            {children}
          </div>
        </div>
        {/* eslint-disable no-use-before-define */}
        {isRefund ? null : renderChange.call(this, optionLen, hasError)}
        {/* eslint-enable */}
      </div>
    );
  }

  roundAmount(amount) {
    const { initialData: { roundCashPaymentType, roundCashPaymentTo } } = this.props;
    const roundedAmount =
      window.roundCashPayments(roundCashPaymentType, roundCashPaymentTo, amount);

    return formatCurrency(roundedAmount);
  }

  onConfirm(pConfirmCallback) {
    pConfirmCallback && typeof pConfirmCallback === 'function' && pConfirmCallback();
  }

  onBlurCashInput(hasError) {
    const amountInput = this._amountInput;
    const fixedAmount = amountInput.value.toFixed(2);
    const { item: { amount } } = this.props;

    (this.roundAmount(amount) !== fixedAmount) && this.changeAmountAndRemaining(hasError);
  }

  changeAmountAndRemaining(hasError) { // calculate remaining and the amount paid when cash allowed
    const { index, item, isRefund, initialData } = this.props;
    const { allowPaymentByCashWithChangeCalculation: allowPayByCash } = initialData;
    const amountInput = this._amountInput;
    let amount = amountInput.value;
    /* istanbul ignore next */
    amount = isNaN(amount) ? 0 : amount;
    let formatCashAmount = item.formatCashAmount;
    let remainingInfo = getRemaining(index, amount, formatCashAmount);
    let alertMessage = '';

    hasError && this.props.clearOptionAndPaymentErrs(index);

    if (typeof remainingInfo === 'number' && remainingInfo < 0) {
      alertMessage = isRefund ?
        fieldProps.MAX_AMOUNT_MESSAGE_ON_REFUND :
        fieldProps.MAX_AMOUNT_MESSAGE_ON_NOT_REFUND;

      this.setState({
        alertMessage
      }, () => {
        this.openConfirm();
      });

      confirmCallback = () => () => {
        amount = +getDefaultAmount(index);
        formatCashAmount = amount.toFixed(2);
        remainingInfo = getRemaining(index, amount, formatCashAmount);
        amountInput.value = this.roundAmount(formatCashAmount);
        this.props.changeCashAmount({ index, amount: formatCashAmount, formatCashAmount });
        this.props.changeRemaining(remainingInfo);

        if (allowPayByCash && !isRefund) {
          const amountPaidInput = this._amountPaidInput;
          this.calculateChange(amountPaidInput.value, formatCashAmount);
        }
      };

      return false;
    }

    formatCashAmount = amount.toFixed(2);
    amountInput.value = this.roundAmount(formatCashAmount);
    this.props.changeCashAmount({ index, amount: formatCashAmount, formatCashAmount });
    this.props.changeRemaining(remainingInfo);

    if (allowPayByCash && !isRefund) {
      const amountPaidInput = this._amountPaidInput;
      this.calculateChange(amountPaidInput.value, formatCashAmount);
    }

    return false;
  }

  openConfirm = () => {
    const option = {
      title: 'System Message',
      className: 'payment-amount-exception-alert',
      confirmText: 'OK'
    };
    return confirm(this.state.alertMessage, option).then(() => {
      this.onConfirm(confirmCallback());
    });
  }

  onBlurAmountPaidInput(value, pFormatCashAmount, hasError) {
    const fixedAmount = value.toFixed(2);
    const { item: { cashAmountPaid } } = this.props;

    this.roundAmount(cashAmountPaid) !== fixedAmount &&
      this.calculateChange(value, pFormatCashAmount, hasError);
  }

  calculateChange(pPaidAmount, pFormatCashAmount, hasError = false) {
    const { index, item } = this.props;
    const autoCalculate = !!pFormatCashAmount;

    const formatCashAmount = typeof pFormatCashAmount === 'undefined' ? item.formatCashAmount : pFormatCashAmount;
    /* istanbul ignore next */
    let paidAmount = isNaN(parseFloat(pPaidAmount)) ? 0 : parseFloat(pPaidAmount);

    if (autoCalculate) {
      paidAmount = formatCashAmount;
    } else if (paidAmount < formatCashAmount) {
      paidAmount = formatCashAmount;
    }

    paidAmount = parseFloat(paidAmount).toFixed(2);
    this._amountPaidInput.value = this.roundAmount(paidAmount);

    hasError && this.props.clearOptionAndPaymentErrs(index);

    this.props.calculateCashChange({
      index,
      change: calculate(paidAmount, formatCashAmount),
      paidAmount
    });
  }
}

function renderChange(optionLen, hasError = false) {
  const className = optionLen > 1 ? 'payment-cash-change-indent' : '';
  const {
    item,
    initialData: { allowPaymentByCashWithChangeCalculation: allowPayByCash }
  } = this.props;

  /* istanbul ignore else */
  if (allowPayByCash) {
    const roundedAmount = this.roundAmount(item.cashAmountPaid);
    const roundValue = this.roundAmount(item.amount);

    const cashChange = formatCurrency(((roundedAmount * 100) - (roundValue * 100)) / 100);

    return (<div className={`payment-cash-change ${className}`}>
      <div className="aaui-flexbox payment-cash-paid">
        <label htmlFor="cashPaid">Amount Paid</label>
        <span className="payment-symbol">$</span>
        <InputNumeric
          id="cashPaid"
          value={roundedAmount}
          ref={(inputNumeric) => { this._amountPaidInput = inputNumeric; }}
          data-qa-id={AMIds.paymentOption.optionAmount}
          min={0}
          onBlur={
            () => this.onBlurAmountPaidInput(this._amountPaidInput.value, undefined, hasError)
          }
        />
      </div>
      <div className="aaui-flexbox">
        {/* eslint-disable jsx-a11y/label-has-for */}
        <label>Change</label>
        {/* eslint-enable */}
        <span className="payment-symbol">$</span>
        <span className="payment-cash-change-amount">{cashChange}</span>
      </div>
      <i className="rectangle" />
    </div>);
  }

  return '';
}

export default connect(
  null,
  {
    changeCashAmount,
    calculateCashChange,
    changeRemaining
  }
)(Cash);
