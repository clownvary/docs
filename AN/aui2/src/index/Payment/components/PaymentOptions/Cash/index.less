@import '~shared/style/variables/index.less';

.payment-cash {
  .aaui-flexbox {
    justify-content: space-between;
  }

  &-change {
    position: relative;
    padding: 18px 10px 10px 0;
    text-align: right;
    background-color: #eef9fd;

    &-indent {
      padding-right: 36px;
    }

    .aaui-flexbox {
      justify-content: flex-end;
    }

    label {
      padding-right: 20px;
      font-size: 16px;
      font-weight: @semibold;
      line-height: 1.5;
    }
  }

  &-change-amount {
    width: 112px;
    text-align: left;
  }

  &-paid {
    margin-bottom: 15px;
  }

  .rectangle {
    position: absolute;
    top: -4px;
    left: 42px;
    width: 8px;
    height: 8px;
    background-color: #eef9fd;
    transform: rotate(45deg);
  }
}
