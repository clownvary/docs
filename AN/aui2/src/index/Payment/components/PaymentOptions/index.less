@import '~shared/style/variables/index.less';

@option-split-border: solid 2px #d6d7da;
@option-split-interval: 18px;
@input-width: 112px;
@font-bold: 600;
/* stylelint-disable */

.payment {
  &-options {
    h2 {
      padding-bottom: 10px;
      margin-top: 0;
      font-size: 20px;
      line-height: 1.5;
      text-transform: capitalize;
      color: #313741;
    }

    .payment {
      &-action {
        justify-content: space-between;
        padding-bottom: 8px;
        font-size: 14px;
      }

      &-distribution {
        .checkbox__text {
          font-size: 14px;
        }
      }

      &-title {
        padding-left: 11px;
        text-transform: uppercase;
        color: #748288;
      }

      &-splition {
        padding-right: 4px;
        cursor: pointer;
        color: @color-blue;

        .icon {
          padding-right: 5px;
        }

        &-disabled {
          cursor: default;
          color: #9b9b9b;

          .icon {
            padding-right: 5px;
            cursor: default;
          }
        }
      }
    }

    .input-group {
      box-sizing: border-box;
      width: @input-width;
    }

    .payment-paynow {
      margin-bottom: 15px;

      .paynow-label {
        display: inline-block;
        width: 114px;
        padding-right: 20px;
        font-weight: @font-bold;
        text-align: right;
      }

      .input-group {
        display: inline-table;
        width: 185px;
        margin-right: 25px;
        vertical-align: middle;
      }

      .paymentplan-label {
        padding-right: 20px;
        font-weight: @font-bold;
      }
    }

    .paynow-alert-error,
    .paymentplan__alert--error {
      .modal-close {
        display: none;
      }
    }
  }

  &-list {
    border-top: @option-split-border;

    & > div:last-child .payment-option {
      margin-bottom: 0;
    }
  }

  &-list-new {
    display: flex;
    flex-direction: column-reverse;
    justify-content: flex-start;
    border-top: @option-split-border;

    &__payment-plan-item {
      flex-grow: 1;
      flex-shrink: 1;
      order: -1;
    }

    &__item {
      flex-grow: 1;
      flex-shrink: 1;
    }
  }

  &-option {
    border-bottom: @option-split-border;

    &-list {
      padding: @option-split-interval 10px 14px 10px;

      &--error {
        z-index: 1;
        background: @state-danger-bg;
        border: 1px solid @state-danger-border;
      }

      &.aaui-flexbox {
        align-items: flex-start;
      }

      .aaui-dropdown {
        width: 170px;
      }
    }

    .icon-trash {
      padding-left: 10px;
      text-align: center;
      cursor: pointer;
      color: @color-blue;

      &-disabled {
        cursor: default;
        color: #9b9b9b;
      }

      &-paymentplan {
        visibility: hidden;
      }
    }
  }

  &-amount-exception-alert {
    .modal-close {
      display: none;
    }
  }

  &-new-cart-msg {
    padding-left: 10px;
    font-size: 12px;
    color: #748288;
  }

  .aaui-flexbox {
    display: flex;
  }
}
