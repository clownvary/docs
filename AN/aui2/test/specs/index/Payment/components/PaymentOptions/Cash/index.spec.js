import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Cash } from 'index/Payment/components/PaymentOptions/Cash';
import InputNumeric from 'react-base-ui/lib/components/InputNumeric';
import Dropdown from 'react-base-ui/lib/components/Dropdown';
import resetActions from 'utils/resetActions';

jest.mock('index/Payment/components/PaymentOptions/utils/payment', () => ({
  getDefaultAmount: jest.fn().mockReturnValue(300),
  getRemaining: jest.fn((index, amount) => amount <= 300 ? -50 : {
    remaining: `$${amount}`
  })
}));

jest.mock('react-base-ui/lib/services/dialog',()=>({
  confirm:jest.fn().mockReturnValue(Promise.resolve(true))
}));

describe('index/Payment/components/PaymentOptions/Cash', () => {
  const actions = {
    changeCashAmount: jest.fn(),
    calculateCashChange: jest.fn(),
    changeRemaining: jest.fn(),
    clearOptionAndPaymentErrs: jest.fn(),
    onChange: jest.fn()
  };

  const initialData = {
    allowPaymentByCashWithChangeCalculation: true
  };

  afterEach(() => {
    resetActions(actions);
  });

  const setup = (props) => mount(<Cash {...props} />);

  it('Cash component should render without error', () => {
    const props = {
      item: {
        list: [
          { value: '172', name: 'Check' },
          { value: '173', name: 'Cash' }
        ],
        amount: 350
      },
      value: '173',
      index: 0,
      optionLen: 1,
      children: (<span>cash info</span>),
      isRefund: false,
      initialData,
      ...actions
    };
    const component = setup(props);
    component.instance().openConfirm();
    expect(toJson(component)).toMatchSnapshot();
  });

  it('Cash component in refund workflow should render without error', () => {
    const props = {
      item: {
        list: [
          { value: '172', name: 'Check' },
          { value: '173', name: 'Cash' }
        ],
        amount: 350,
        cashAmountPaid: 100,
        cashChange: 100
      },
      value: '173',
      index: 0,
      optionLen: 3,
      children: (<span>cash info</span>),
      isRefund: true,
      initialData,
      ...actions
    };
    const component = setup(props);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('Cash component payment process should work fine', () => {
    const props = {
      item: {
        list: [
          { value: '172', name: 'Check' },
          { value: '173', name: 'Cash' }
        ],
        amount: 0,
        formatCashAmount: 300
      },
      value: '173',
      index: 2,
      optionLen: 3,
      children: (<span>cash info</span>),
      isRefund: false,
      initialData,
      ...actions
    };

    const component = setup(props);
    const _ins = component.instance();
    const confirmMock = jest.spyOn(_ins,'openConfirm');
    const changeAmountAndRemainingMock = jest.spyOn(_ins,'changeAmountAndRemaining');

    expect(component.find('div.payment-option-list')).toHaveLength(1);
    expect(component.find('#cashAmount')).toHaveLength(1);
    expect(component.find('div.payment-cash-change.payment-cash-change-indent')).toHaveLength(1);
    expect(component.find('#cashPaid')).toHaveLength(1);

    const paymentOptionDropdown = component.find(Dropdown);
    expect(paymentOptionDropdown).toHaveLength(1);
    paymentOptionDropdown.node.props.onChange({});
    expect(actions.onChange).toBeCalled();

    const amountInput = component.find(InputNumeric).at(0);
    const amountInputHtmlElement = component.find('#cashAmount');
    expect(amountInputHtmlElement).toHaveLength(1);

    amountInputHtmlElement.simulate('blur');
    expect(changeAmountAndRemainingMock).not.toHaveBeenCalled();

    amountInput.node.input.value = 200;
    amountInputHtmlElement.simulate('blur');
    expect(confirmMock).toHaveBeenCalled();
    component.instance().openConfirm().then(()=>{
      expect(actions.changeCashAmount).toHaveBeenCalled();
      expect(actions.changeRemaining).toHaveBeenCalled();
      expect(actions.calculateCashChange).toHaveBeenCalled();
    });

    const amountPaidInputHtmlElement = component.find('#cashPaid');
    const amountPaidInput = component.find(InputNumeric).at(1);
    amountPaidInput.node.input.value = 100;
    amountPaidInputHtmlElement.simulate('blur');
    component.instance().openConfirm().then(()=>{
      expect(actions.calculateCashChange).toHaveBeenCalled();
    });

    component.setProps({ item: { errors: [{ name: '' }] } });
    amountInput.node.input.value = 100;
    amountInputHtmlElement.simulate('blur');
    expect(actions.clearOptionAndPaymentErrs).toHaveBeenCalled();

  });

  it('Cash component payment process should work fine if meets error', () => {
    const props = {
      item: {
        list: [
          { value: '172', name: 'Check' },
          { value: '173', name: 'Cash' }
        ],
        amount: 0,
        formatCashAmount: 300,
        errors: [{ message: 'cash mock error', name: '' }]
      },
      value: '173',
      index: 2,
      optionLen: 3,
      children: (<span>cash info</span>),
      isRefund: false,
      initialData,
      ...actions
    };

    const component = setup(props);

    expect(component.find('div.payment-option-list')).toHaveLength(1);
    expect(component.find('#cashAmount')).toHaveLength(1);
    expect(component.find('div.payment-cash-change.payment-cash-change-indent')).toHaveLength(1);
    expect(component.find('#cashPaid')).toHaveLength(1);
    expect(component.find('.payment-option-list--error')).toHaveLength(1);
    expect(component.find('.payment-cash-change')).toHaveLength(1);

    const amountInput = component.find(InputNumeric).at(0);
    const amountInputHtmlElement = component.find('#cashAmount');
    expect(amountInputHtmlElement).toHaveLength(1);

    amountInput.node.input.value = 200;
    amountInputHtmlElement.simulate('blur');
    expect(actions.clearOptionAndPaymentErrs).toHaveBeenCalledTimes(1);

    const amountPaidInputHtmlElement = component.find('#cashPaid');
    const amountPaidInput = component.find(InputNumeric).at(1);
    amountPaidInput.node.input.value = 100;
    amountPaidInputHtmlElement.simulate('blur');
    expect(actions.clearOptionAndPaymentErrs).toHaveBeenCalledTimes(2);
  });

  it('Cash component refund process should work fine', () => {
    const props = {
      item: {
        list: [
          { value: '172', name: 'Check' },
          { value: '173', name: 'Cash' }
        ],
        amount: 350,
        formatCashAmount: 350
      },
      value: '173',
      index: 0,
      optionLen: 1,
      children: (<span>cash info</span>),
      isRefund: true,
      initialData,
      ...actions
    };

    const component = setup(props);
    const _ins = component.instance();
    const confirmMock = jest.spyOn(_ins,'openConfirm');
    expect(component.find('div.payment-option-list')).toHaveLength(1);
    expect(component.find('#cashAmount')).toHaveLength(1);
    expect(component.find('div.payment-cash-change.payment-cash-change-indent')).toHaveLength(0);
    expect(component.find('#cashPaid')).toHaveLength(0);

    const amountInput = component.find(InputNumeric).at(0);
    const amountInputHtmlElement = component.find('#cashAmount');
    expect(amountInputHtmlElement).toHaveLength(1);

    amountInput.node.input.value = 200;
    amountInputHtmlElement.simulate('blur');
    expect(confirmMock).toHaveBeenCalled();
    component.instance().openConfirm().then(()=>{
      expect(actions.changeCashAmount).toHaveBeenCalled();
      expect(actions.changeRemaining).toHaveBeenCalled();
      expect(actions.calculateCashChange).not.toHaveBeenCalled();
    });

    amountInput.node.input.value = 1200;
    amountInputHtmlElement.simulate('blur');
    component.instance().openConfirm().then(()=>{
      expect(actions.changeCashAmount).toHaveBeenCalled();
      expect(actions.changeRemaining).toHaveBeenCalled();
      expect(actions.calculateCashChange).not.toHaveBeenCalled();
    });
  });
  it('Cash component should work fine when not refund',()=>{
    const props = {
      item: {
        list: [
          { value: '172', name: 'Check' },
          { value: '173', name: 'Cash' }
        ],
        amount: 350,
        formatCashAmount: 350
      },
      value: '173',
      index: 0,
      optionLen: 1,
      children: (<span>cash info</span>),
      isRefund: false,
      initialData,
      ...actions
    };

    const component = setup(props);
    const amountInputHtmlElement = component.find('#cashAmount');
    const amountInput = component.find(InputNumeric).at(0);
    expect(amountInputHtmlElement).toHaveLength(1)
    amountInput.node.input.value = 1200;
    amountInputHtmlElement.simulate('blur');
    component.instance().openConfirm().then(()=>{
      expect(actions.changeCashAmount).toHaveBeenCalled();
      expect(actions.changeRemaining).toHaveBeenCalled();
      expect(actions.calculateCashChange).toHaveBeenCalled();
    });
  });
});
