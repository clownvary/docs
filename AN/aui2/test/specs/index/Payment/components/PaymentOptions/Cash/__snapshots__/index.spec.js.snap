// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index/Payment/components/PaymentOptions/Cash Cash component in refund workflow should render without error 1`] = `
<Cash
  calculateCashChange={[Function]}
  changeCashAmount={[Function]}
  changeRemaining={[Function]}
  clearOptionAndPaymentErrs={[Function]}
  index={0}
  initialData={
    Object {
      "allowPaymentByCashWithChangeCalculation": true,
    }
  }
  isRefund={true}
  item={
    Object {
      "amount": 350,
      "cashAmountPaid": 100,
      "cashChange": 100,
      "list": Array [
        Object {
          "name": "Check",
          "value": "172",
        },
        Object {
          "name": "Cash",
          "value": "173",
        },
      ],
    }
  }
  onChange={[Function]}
  optionLen={3}
  value="173"
>
  <div
    className="payment-option payment-cash"
  >
    <div
      className="payment-option-list aaui-flexbox"
    >
      <Dropdown
        autoOpen={false}
        data={
          Array [
            Object {
              "name": "Check",
              "value": "172",
            },
            Object {
              "name": "Cash",
              "value": "173",
            },
          ]
        }
        data-qa-id="payment-paymentOption-optionList"
        filterPlaceholder="Filter..."
        fuzzyQuery={false}
        noResult="No results found."
        onChange={[Function]}
        placeholder="Select one..."
        showAll={true}
        showDeselectall={false}
        showTxtOnlyCheckedOneItem={true}
        theme="flat"
        txtSuffix="selected"
        value="173"
        visible={true}
      >
        <div
          className="aaui-dropdown"
          data-qa-id="payment-paymentOption-optionList"
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onKeyUp={[Function]}
        >
          <button
            className="aaui-dropdown__button aaui-dropdown-btn-flat"
            onClick={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
            role="button"
            type="button"
          >
            <div
              className="aaui-dropdown__button-text"
              title=""
            >
               Select one...
            </div>
            <i
              className="icon aaui-dropdown__button-icon icon-chevron-down"
            />
          </button>
          <div
            aria-expanded={false}
            className="aaui-dropdown__menu-wrapper  u-hidden"
            onBlur={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            role="combobox"
            tabIndex={0}
          >
            <div
              className="aaui-dropdown__menu"
            >
              <ul
                aria-expanded={false}
                className=""
                role="listbox"
              >
                <li
                  className=""
                  key="172"
                  onClick={[Function]}
                  role="option"
                >
                  <span />
                </li>
                <li
                  className=" is-selected"
                  key="173"
                  onClick={[Function]}
                  role="option"
                >
                  <span />
                </li>
              </ul>
              <div>
                <div
                  className="aaui-dropdown__menu-footer"
                >
                  <div
                    className="u-hidden"
                  >
                    <i
                      className="icon icon-spinner"
                    />
                     Loading...
                  </div>
                  <div />
                </div>
              </div>
            </div>
          </div>
        </div>
      </Dropdown>
      <div
        className="aaui-flexbox"
      >
        <label
          className="payment-symbol"
          htmlFor="cashAmount"
        >
          $
        </label>
        <InputNumeric
          allowBlank={true}
          allowKeySpin={true}
          allowMouseSpin={false}
          className=""
          currency="USD"
          data-qa-id="payment-paymentOption-optionAmount"
          decimals={2}
          disabled={false}
          icon=""
          iconHint=""
          id="cashAmount"
          increment={1}
          items={Array []}
          listClass="input-numeric-list"
          listClassName=""
          listConfig={Object {}}
          listMaxHeight="300px"
          listPopupOptions={Object {}}
          listWidth="auto"
          max={9999999999.99}
          min={0}
          name=""
          onBlur={[Function]}
          onFocus={[Function]}
          onListClose={[Function]}
          onListOpen={[Function]}
          onTextChange={[Function]}
          onValueChange={[Function]}
          placeHolder=""
          prefix="an"
          showClear={false}
          showGroup={false}
          showSpinner={false}
          showTrigger={false}
          showTrigger2={false}
          size="md"
          textAlign="right"
          triggerIcon="icon-chevron-down"
          triggerIcon2="icon-chevron-down"
          triggerIconHint=""
          triggerIconHint2=""
          triggerIconToggle="icon-chevron-up"
          triggerIconToggle2="icon-chevron-up"
          value="0.00"
        >
          <div
            className="an-input-numeric input-group buttons-0 input-group--md"
          >
            <input
              className="input input-group__field textalign-right input__field"
              data-qa-id="payment-paymentOption-optionAmount"
              disabled={false}
              id="cashAmount"
              name=""
              onBlur={[Function]}
              onChange={[Function]}
              onDragEnter={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onKeyUp={[Function]}
              onMouseUp={[Function]}
              onPaste={[Function]}
              placeholder=""
              type="text"
            />
          </div>
        </InputNumeric>
        <span>
          cash info
        </span>
      </div>
    </div>
  </div>
</Cash>
`;

exports[`index/Payment/components/PaymentOptions/Cash Cash component should render without error 1`] = `
<Cash
  calculateCashChange={[Function]}
  changeCashAmount={[Function]}
  changeRemaining={[Function]}
  clearOptionAndPaymentErrs={[Function]}
  index={0}
  initialData={
    Object {
      "allowPaymentByCashWithChangeCalculation": true,
    }
  }
  isRefund={false}
  item={
    Object {
      "amount": 350,
      "list": Array [
        Object {
          "name": "Check",
          "value": "172",
        },
        Object {
          "name": "Cash",
          "value": "173",
        },
      ],
    }
  }
  onChange={[Function]}
  optionLen={1}
  value="173"
>
  <div
    className="payment-option payment-cash"
  >
    <div
      className="payment-option-list aaui-flexbox"
    >
      <Dropdown
        autoOpen={false}
        data={
          Array [
            Object {
              "name": "Check",
              "value": "172",
            },
            Object {
              "name": "Cash",
              "value": "173",
            },
          ]
        }
        data-qa-id="payment-paymentOption-optionList"
        disabled={false}
        filterPlaceholder="Filter..."
        fuzzyQuery={false}
        noResult="No results found."
        onChange={[Function]}
        placeholder="Select one..."
        showAll={true}
        showDeselectall={false}
        showTxtOnlyCheckedOneItem={true}
        theme="flat"
        txtSuffix="selected"
        value="173"
        visible={true}
      >
        <div
          className="aaui-dropdown"
          data-qa-id="payment-paymentOption-optionList"
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onKeyUp={[Function]}
        >
          <button
            className="aaui-dropdown__button aaui-dropdown-btn-flat"
            onClick={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
            role="button"
            type="button"
          >
            <div
              className="aaui-dropdown__button-text"
              title=""
            >
               Select one...
            </div>
            <i
              className="icon aaui-dropdown__button-icon icon-chevron-down"
            />
          </button>
          <div
            aria-expanded={false}
            className="aaui-dropdown__menu-wrapper  u-hidden"
            onBlur={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            role="combobox"
            tabIndex={0}
          >
            <div
              className="aaui-dropdown__menu"
            >
              <ul
                aria-expanded={false}
                className=""
                role="listbox"
              >
                <li
                  className=""
                  key="172"
                  onClick={[Function]}
                  role="option"
                >
                  <span />
                </li>
                <li
                  className=" is-selected"
                  key="173"
                  onClick={[Function]}
                  role="option"
                >
                  <span />
                </li>
              </ul>
              <div>
                <div
                  className="aaui-dropdown__menu-footer"
                >
                  <div
                    className="u-hidden"
                  >
                    <i
                      className="icon icon-spinner"
                    />
                     Loading...
                  </div>
                  <div />
                </div>
              </div>
            </div>
          </div>
        </div>
      </Dropdown>
      <div
        className="aaui-flexbox"
      >
        <label
          className="payment-symbol"
          htmlFor="cashAmount"
        >
          $
        </label>
        <InputNumeric
          allowBlank={true}
          allowKeySpin={true}
          allowMouseSpin={false}
          className=""
          currency="USD"
          data-qa-id="payment-paymentOption-optionAmount"
          decimals={2}
          disabled={false}
          icon=""
          iconHint=""
          id="cashAmount"
          increment={1}
          items={Array []}
          listClass="input-numeric-list"
          listClassName=""
          listConfig={Object {}}
          listMaxHeight="300px"
          listPopupOptions={Object {}}
          listWidth="auto"
          max={9999999999.99}
          min={0}
          name=""
          onBlur={[Function]}
          onFocus={[Function]}
          onListClose={[Function]}
          onListOpen={[Function]}
          onTextChange={[Function]}
          onValueChange={[Function]}
          placeHolder=""
          prefix="an"
          showClear={false}
          showGroup={false}
          showSpinner={false}
          showTrigger={false}
          showTrigger2={false}
          size="md"
          textAlign="right"
          triggerIcon="icon-chevron-down"
          triggerIcon2="icon-chevron-down"
          triggerIconHint=""
          triggerIconHint2=""
          triggerIconToggle="icon-chevron-up"
          triggerIconToggle2="icon-chevron-up"
          value="0.00"
        >
          <div
            className="an-input-numeric input-group buttons-0 input-group--md"
          >
            <input
              className="input input-group__field textalign-right input__field"
              data-qa-id="payment-paymentOption-optionAmount"
              disabled={false}
              id="cashAmount"
              name=""
              onBlur={[Function]}
              onChange={[Function]}
              onDragEnter={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onKeyUp={[Function]}
              onMouseUp={[Function]}
              onPaste={[Function]}
              placeholder=""
              type="text"
            />
          </div>
        </InputNumeric>
        <span>
          cash info
        </span>
      </div>
    </div>
    <div
      className="payment-cash-change "
    >
      <div
        className="aaui-flexbox payment-cash-paid"
      >
        <label
          htmlFor="cashPaid"
        >
          Amount Paid
        </label>
        <span
          className="payment-symbol"
        >
          $
        </span>
        <InputNumeric
          allowBlank={true}
          allowKeySpin={true}
          allowMouseSpin={false}
          className=""
          currency="USD"
          data-qa-id="payment-paymentOption-optionAmount"
          decimals={2}
          disabled={false}
          icon=""
          iconHint=""
          id="cashPaid"
          increment={1}
          items={Array []}
          listClass="input-numeric-list"
          listClassName=""
          listConfig={Object {}}
          listMaxHeight="300px"
          listPopupOptions={Object {}}
          listWidth="auto"
          max={9999999999.99}
          min={0}
          name=""
          onBlur={[Function]}
          onFocus={[Function]}
          onListClose={[Function]}
          onListOpen={[Function]}
          onTextChange={[Function]}
          onValueChange={[Function]}
          placeHolder=""
          prefix="an"
          showClear={false}
          showGroup={false}
          showSpinner={false}
          showTrigger={false}
          showTrigger2={false}
          size="md"
          textAlign="right"
          triggerIcon="icon-chevron-down"
          triggerIcon2="icon-chevron-down"
          triggerIconHint=""
          triggerIconHint2=""
          triggerIconToggle="icon-chevron-up"
          triggerIconToggle2="icon-chevron-up"
          value="0.00"
        >
          <div
            className="an-input-numeric input-group buttons-0 input-group--md"
          >
            <input
              className="input input-group__field textalign-right input__field"
              data-qa-id="payment-paymentOption-optionAmount"
              disabled={false}
              id="cashPaid"
              name=""
              onBlur={[Function]}
              onChange={[Function]}
              onDragEnter={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onKeyUp={[Function]}
              onMouseUp={[Function]}
              onPaste={[Function]}
              placeholder=""
              type="text"
            />
          </div>
        </InputNumeric>
      </div>
      <div
        className="aaui-flexbox"
      >
        <label>
          Change
        </label>
        <span
          className="payment-symbol"
        >
          $
        </span>
        <span
          className="payment-cash-change-amount"
        >
          0.00
        </span>
      </div>
      <i
        className="rectangle"
      />
    </div>
  </div>
</Cash>
`;
